coredns/plugin/go.mod

```diff
@@ -0,0 +1,84 @@
+module github.com/kuadrant/coredns-kuadrant
+
+go 1.23.0
+
+toolchain go1.23.4
```

Comment:
Hmm, kind of wondering if we should just set this back to 1.22 :thinking: It should probably be the same as what CoreDNS currently set anyway https://github.com/coredns/coredns/blob/master/go.mod#L3

Reply:
Yeah I tihnk this would make sense for now

Reply:
I have set this back to 1.22.7. I'm unsure about the best way forward here, ideally we wouldn't be restricted in updating things because of downstream build limitations. CoreDNS now set the toolchain to 1.23+ https://github.com/coredns/coredns/blob/master/go.mod#L3-L5 which allows the dependencies with issues to update.

Will come back to this.